### PR TITLE
Added app router and user bloc 

### PR DESCRIPTION
**Pull request by ehsankkk1**

### CODE REVIEW

**Review by samuelkchris — DISMISSED** [review]
so far we look good

File: android/app/src/main/java/io/flutter/app/FlutterMultiDexApplication.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import android.app.Application;
+import android.content.Context;
+import androidx.annotation.CallSuper;
+import androidx.multidex.MultiDex;
+
+/**
+ * Extension of {@link android.app.Application}, adding multidex support.
+ */
+public class FlutterMultiDexApplication extends Application {
```
Comment (BurningAXE) [inline android/app/src/main/java/io/flutter/app/FlutterMultiDexApplication.java:18]: Do we need multi DEX support?

File: lib/src/services/constants.dart [hunk 2]
```diff
@@ -1,3 +1,5 @@
+
+// todo : change to app router method in utility folder
```
Comment (BurningAXE) [inline lib/src/services/constants.dart]: Yes, please move the constants to the app router

File: lib/utility/app_router.dart [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    return MaterialPageRoute(builder: (context) {
+
+      // add new case with the name of your route and the screen you will navigate to
+
+      return Builder(
+        builder: (BuildContext context) {
+          switch (settings.name) {
+
+            case AppRoutes.homeScreen:
+              return const HomeScreen(title: 'Help',);
```
Comment (BurningAXE) [inline lib/utility/app_router.dart]: 'Help a Paw' 😇

File: lib/utility/app_router.dart [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+
+
+              // in case the named route provided in not available
+            default:
```
Comment (BurningAXE) [inline lib/utility/app_router.dart:34]: Maybe default to home & show a snackbar "Destination not found"?

File: linux/flutter/generated_plugin_registrant.cc
Comment (BurningAXE) [inline linux/flutter/generated_plugin_registrant.cc:1]: Why did we get generated files for linux, macos & windows? We'll only use iOS, Android and possibly web so better not have the rest.

**Review by BurningAXE — CHANGES_REQUESTED** [review]
Very good job, man! Requesting just some minor changes.

File: android/app/src/main/java/io/flutter/app/FlutterMultiDexApplication.java (hunk 1 above)
Comment (ehsankkk1) [reply android/app/src/main/java/io/flutter/app/FlutterMultiDexApplication.java:18]: can't run the app without it  can't figure why

File: lib/src/services/constants.dart (hunk 2 above)
Comment (ehsankkk1) [reply lib/src/services/constants.dart]: done

File: lib/utility/app_router.dart (hunk 4 above)
Comment (ehsankkk1) [reply lib/utility/app_router.dart:34]: I thinks this method is better for debugging when going full release we need to change to your method .

**Review by ehsankkk1 — COMMENTED** [review]
all ready to merge

File: lib/src/pages/splash.dart [hunk 5]
```diff
@@ -16,12 +17,8 @@ class SplashScreen extends StatelessWidget {
...
     Future.delayed(const Duration(seconds: 2), () async {
       bool isLoggedIn = await UserValidation().checkUserLogin();
-      Navigator.of(context).push(
-        MaterialPageRoute(
-            builder: (context) => const HomePage(title: 'Help a Paw'),
-            settings: RouteSettings(
-                name: isLoggedIn ? global.homeRoute : global.loginRoute)),
-      );
+
+      Navigator.of(context).pushNamed(isLoggedIn ? AppRoutes.homeScreen : AppRoutes.loginRoute);
```
Comment (BurningAXE) [inline lib/src/pages/splash.dart]: Actually we want to always lend on the home screen and not force users to login.

**Review by BurningAXE — CHANGES_REQUESTED** [review]
Good job! Just one small change about the Splash screen

File: lib/src/pages/splash.dart (hunk 5 above)
Comment (ehsankkk1) [reply lib/src/pages/splash.dart]: done

**Review by BurningAXE — APPROVED** [review]

**Pull request merged**